unikube/graphql_utils.py

```diff
@@ -72,23 +72,23 @@ def query(
         query: str,
         query_variables: dict = None,
     ) -> Union[dict, None]:
-        with click_spinner.spinner(beep=False, disable=False, force=False, stream=sys.stdout):
-            try:
-                query = gql(query)
+        try:
+            query = gql(query)
+            with click_spinner.spinner(beep=False, disable=False, force=False, stream=sys.stdout):
                 data = self.client.execute(
                     document=query,
                     variable_values=query_variables,
                 )
 
-            except requests.exceptions.HTTPError:
-                # refresh token
-                response = self.authentication.refresh()
-                if not response["success"]:
-                    console.exit_login_required()
+        except TransportServerError:
```

Comment:
@SteinRobert Could this be even more specific like testing for the status code? It feels like there may be other errors possible that will trigger a login attempt. But sInce the missing login is by far the most common in what we have observed, I would go ahead with this but keep an open eye for such other cases.

Reply:
Totally makes sense. Maybe we should create an issue for the other cases?